File: AOJ/0316/ori/0316.cpp
#include<iostream>
using namespace std;

main(){
  int h1, h2, k1, k2, a, b, c, d;
  cin >> h1 >> h2 >> k1 >> k2;
  cin >> a >> b >> c >> d;
  int hiroshi = h1*a + h2*b + (h1/10)*c + (h2/20)*d;
  int kenjiro = k1*a + k2*b + (k1/10)*c + (k2/20)*d;

  if ( hiroshi > kenjiro ) cout << "hiroshi" << endl;
  else if ( hiroshi < kenjiro ) cout << "kenjiro" << endl;
  else cout << "even" << endl;
}

File: AOJ/0316/ori/0316.py
h1, h2 = map(int, input().split())
k1, k2 = map(int, input().split())
a, b, c, d = map(int, input().split())
hiroshi = h1*a + h2*b + (h1//10)*c + (h2//20)*d;
kenjiro = k1*a + k2*b + (k1//10)*c + (k2//20)*d;

if hiroshi > kenjiro: print("hiroshi")
elif hiroshi < kenjiro: print("kenjiro")
else: print("even")
